sdk/client/src/stream/streaming_client.rs

```diff
@@ -0,0 +1,204 @@
+        let id = match &msg.id {
+            Some(id) => id,
+            None => {
+                self.error_sender.send(Ok(msg)).await.ok();
+                return Ok(());
+            }
+        };
+        // Send the message to the respective channel
+        let id = id.clone();
+        match self.subscriptions.read().await.get(&id) {
+            // If we could not send the subscription, or the channel is closed, make sure to clean up the subscription
+            Some(sender) => match sender.send(Ok(msg)).await {
+                Err(e) => {
+                    self.unsubscribe(&id).await;
+                    Err(Error::connection_closed(Some(e)))
+                }
+                Ok(_) => Ok(()),
+            },
+            // No such subscription exists
+            None => {
+                self.error_sender
+                    .send(Err(Error::subscription_id_not_found(
+                        Some(msg),
+                        None::<Error>,
+                    )))
+                    .await
+                    .ok();
+                self.unsubscribe(&id).await;
+                Ok(())
+            }
+        }
+    }
+
+    pub async fn unsubscribe(&self, id: &Id) {
+        self.subscriptions.write().await.remove(id);
+    }
+}
+
+pub struct StreamingClient<C: StreamingClientTransport + 'static> {
+    client: Arc<RwLock<C>>,
+    subscriptions: Arc<RwLock<HashMap<Id, StreamingClientSender>>>,
+    channel_size: usize,
+    error_sender: StreamingClientSender,
+    channel_task: Option<JoinHandle<Result<()>>>,
+}
+
+impl<C: StreamingClientTransport + 'static> StreamingClient<C> {
+    pub async fn new(client: C, channel_size: usize) -> (Self, StreamingClientReceiver) {
+        let subscriptions = Arc::new(RwLock::new(HashMap::new()));
+        let (error_sender, error_receiver) =
+            mpsc::channel::<Result<StreamJsonRpcResponse>>(channel_size);
+
+        subscriptions
+            .write()
+            .await
+            .insert(Id::String(Box::from("_errors")), error_sender.clone());
+
+        let (stream, client) = client.get_stream();
+
+        let mut sct = Self {
+            client: Arc::new(RwLock::new(client)),
+            subscriptions,
+            channel_size,
+            error_sender,
+            channel_task: None,
+        };
+
+        sct.start_channel_task(stream);
+
+        (sct, error_receiver)
+    }
+
+    #[allow(unused)]
+    async fn subscribe_transactions(
+        &mut self,
+        starting_version: u64,
+        include_events: Option<bool>,
+    ) -> SubscriptionStreamResult {
+        let (id, receiver) = self.get_and_register_id().await?;
+        let res = self
+            .client
+            .write()
+            .await
+            .subscribe_transactions(starting_version, include_events, Some(id.clone()))
+            .await;
+
+        let id = self.maybe_clear_subscription(&id, res).await?;
+        Ok((id, receiver))
+    }
+
+    #[allow(unused)]
```

Comment:
Why are there a few things marked as unused?

Reply:
because they're not currenty used outside of smoke tests and clippy was yelling at me

Reply:
Same thing here